doc/python_api_reference_vDev.md

```diff
@@ -7793,6 +7837,93 @@ def to_unitary_matrix(
     """
 ```
 
+<a name="stim.PauliStringIterator"></a>
+```python
+# stim.PauliStringIterator
+
+# (at top-level in the stim module)
+class PauliStringIterator:
+    """Iterates over all possible pauli_strings of weight specfied by
+    min_weight and max_weight.
+
+    Returns:
+        An Iterable[stim.PauliString] that yields the requested Pauli
+            string.
+
+    Examples:
+        >>> import stim
+        >>> pauli_iter = stim.PauliString.iter_all(10, 2, 3)
+        >>> n = 0
+        >>> for pauli_string in pauli_iter:
+        ...     n += 1
+        >>> n
+    """
+```
+
+<a name="stim.PauliStringIterator.__iter__"></a>
+```python
+# stim.PauliStringIterator.__iter__
+
+# (in class stim.PauliStringIterator)
+def __iter__(
+    self,
+) -> stim.PauliStringIterator:
+    """Returns an independent copy of the pauli string iterator.
+    """
+```
+
+<a name="stim.PauliStringIterator.__next__"></a>
+```python
+# stim.PauliStringIterator.__next__
+
+# (in class stim.PauliStringIterator)
+def __next__(
+    self,
+) -> stim.PauliString:
+    """Returns the next iterated pauli string.
+    """
+```
+
+
+<a name="stim.PauliStringIterator.next_qubit_permutation"></a>
+```python
+# stim.PauliStringIterator.next_qubit_permutation
+
+# (in class stim.PauliStringIterator)
+def next_qubit_permutation(
+    self,
+) -> object:
+    """Get the next permutation of qubit labels.
+
+    It's alot easier to test more complicated edge cases in python
+    which largely arise due to the algorithm for generating the next
+    permutation. The user should not interact with this function.
```

Comment:
If the user shouldn't interact with the function, start its name with something like `__internal_` and throw in a `@deprecated` in the docstring so it doesn't show in the api reference.